### DISCUSSION TITLE
Isar + StateManagement or just have UI watch/write to Isar directly?

### DISCUSSION THREAD

**Discussion by Leffe108** [isar/isar#259] [Help]
Isar looks interesting. In the sample app you seem to make it reactive by means of watching the database and stream builder.

Is that the recommended route for reactive UI, or would one in some cases still use a StateManagement package like Provider and make the provider model write to Isar and listen to Isar using (lazy)watch and then notifiyListeners()? But maybe that is an anti-parttern and it is best to just drop provider and have UI react directly on Isar lazy watch and watch? And wire up user input to database writes.

**Comment by simc** [comment]
It very much depends on your use case. I plan to create more advanced example apps but generally it is not an anti pattern because watchers are quite efficient and have little overhead.

But Isar is NOT a general replacement for state management.

The only thing you should be careful about is non-lazy query watchers. Only use them if you really need to re-run the query for every change.
